### DISCUSSION TITLE
Strange Error in Address

### DISCUSSION THREAD

**Discussion by rengit62** [OP-TED/eForms-SDK#654] [Q&A] [answered]
Hi Lab !
If I miss street_name in Procedure I get a clear error : 

![image](https://github.com/OP-TED/eForms-SDK/assets/52211037/c25ccfd6-eaa0-453d-ae6e-2213ba238bbb)

in log : 
id : BR-BT-05101-0063
location : /pin:PriorInformationNotice/cac:ProcurementProject/cac:RealizedLocation/cac:Address
failedAssertText : BT-5101(b)-Procedure is not allowed in notice type '7' under following condition: 'Postal address' (BT-5101(a)-Procedure) is not filled in
diagnostic : BT-5101_b_-Procedure
see : field:BT-5101(b)-Procedure
diagnoticReferenceText : cbc:AdditionalStreetName

If I miss street name in Organisations Adress 
![image](https://github.com/OP-TED/eForms-SDK/assets/52211037/ed6f773e-8914-4cd0-b688-61bd05038d37)

The error log is 'poor' ..
id : element-not-empty
location : /pin:PriorInformationNotice/ext:UBLExtensions/ext:UBLExtension/ext:ExtensionContent/efext:EformsExtension/efac:Organizations/efac:Organization[1]/efac:Company/cac:PostalAddress/cbc:StreetName
failedAssertText : The fields must be filled in

it is not the same error with different fields ?
thanks and regards 
Renato

**Comment by YvesJo** [comment, marked as answer]
Hi,
Indeed it is not the same error with different fields.
The rules are context dependent and the conditions may differ. An organization has to have an official physical address while a place of performance may refer to a local area only.
Also, given the context dependency, different rule messages have to be defined.
Regarding the specific error message for the organization address, the rule that generated it is not the same as for the previous case, it is related to a generic rule checking that no XML element is empty; so, most likely, the generated XML contains an empty cbc:StreetName element for that organization.

**Reply by rengit62** [reply]
OK, forget what I wrote about 'same error'.
Anyway the first  validation message (id : BR-BT-05101-0063) show the id of the field that must be filled in, the second does not .
It is possible modify the rule (id : element-not-empty)  to show the field id?
Thanks and Regards

**Reply by YvesJo** [reply]
There is indeed a need to check whether this would be possible; I can't tell whether this is feasible and will raise that request.

**Reply by YvesJo** [reply]
Hi,
Following internal analysis, this won't be feasible without introducing more than a thousand asserts, which would be inappropriate. 
Also, an end user should never face such a situation as the solution in place should never output the element when its content is empty (or just filled with whitespaces).
KR
